Fix download func and Change class SkinCancerMnist

dinhanhx opened the pull request:
Demo new func 
```Python
import json
import kaggle

dataset = 'dinhanhx/skin-cancer-mnist-ham10000'
dataset_path = kaggle.download(dataset)
scm = kaggle.SkinCancerMnist(dataset_path)
print(json.dumps(scm.get_img_metadat(0), indent=3))
```
Output
```
{
   "Image path": "C:\\Users\\DinhAnh\\.kaggle\\dinhanhx\\skin-cancer-mnist-ham10000\\ISIC_0024306.jpg",
   "Host": "Kaggle",
   "Original url": "https://www.kaggle.com/dinhanhx/skin-cancer-mnist-ham10000",
   "Caption": "HAM_0000550"
}
```

dinhanhx committed:
Add validation function

dinhanhx committed:
Add kaggle==1.6.9

dinhanhx committed:
Reformat code

dinhanhx committed:
Update docstring

dinhanhx committed:
Reformat docstring of validate()

dinhanhx committed:
Add download() for datasets manager

dinhanhx committed:
Add class skin_cancer_mnist

dinhanhx committed:
Add download(), class skin_cancer_mnist

dinhanhx committed:
Add kaggle documentation

dinhanhx committed:
Ignore __init__.py files

dinhanhx committed:
Add get_dataset_url() to class skin_cancer_mnist

dinhanhx committed:
I hate flake8

dinhanhx committed:
Rewrite doc to be more human friendly

dinhanhx committed:
Rewrite docstrings

dinhanhx committed:
Rewrite docstrings

dinhanhx committed:
Add 2 functions of class skin_cancer_mnist

dinhanhx committed:
Add .flake8 for configuration

dinhanhx committed:
Move .flake8 to root

dinhanhx committed:
Split doc into user and module

dinhanhx committed:
Update backend/kaggle/datasets_manager.py

Co-authored-by: Ngô Ngọc Đức Huy <duchuy29092000@gmail.com>

dinhanhx committed:
Rename skin_cancer_mnist to SkinCancerMnist

dinhanhx committed:
Rename skin_cancer_mnist to SkinCancerMnist

dinhanhx committed:
Merge remote-tracking branch 'upstream/main' into main

dinhanhx committed:
Merge remote-tracking branch 'upstream/main' into main

dinhanhx committed:
func download and class SkinCancerMnist
func download now returns full path even dataset exists
class SkinCancerMnist now processes dataset from dinhanhx

dinhanhx committed:
Add func export_img_to_flask() for class SkinCancerMnist
Also add to kaggle documentation

Huy-Ngo commented on backend/kaggle/README.md:
````diff
@@ -19,7 +19,7 @@ if kaggle.validate():
 ```
 
-Firstly, it checks for kaggle api and kaggle package. Then it downloads the dataset then save it this directory `~/.kaggle/kmader/skin-cancer-mnist-ham10000`. For Windows, it's `YourUserName/.kaggle/kmader/skin-cancer-mnist-ham10000`. Secondly, it creates an object to manipulate the dataset. Finally it prints to stdout what that object has.
+Firstly, it checks for kaggle api and kaggle package. Then it downloads the dataset then save it this directory `~/.kaggle/dinhanhx/skin-cancer-mnist-ham10000`. For Windows, it's `YourUserName/.kaggle/dinhanhx/skin-cancer-mnist-ham10000`. Secondly, it creates an object to manipulate the dataset. Finally it prints to stdout what that object has.
````
```suggestion
Firstly, it checks for kaggle api and kaggle package. Then it downloads the dataset then save it to directory `~/.kaggle/dinhanhx/skin-cancer-mnist-ham10000`. For Windows, it's `YourUserName/.kaggle/dinhanhx/skin-cancer-mnist-ham10000`. Secondly, it creates an object to manipulate the dataset. Finally it prints to stdout what that object has.
```

Huy-Ngo commented on backend/kaggle/README.md:
````diff
@@ -6,11 +6,11 @@ import kaggle
 ```Python
 if kaggle.validate():
-  dataset = 'kmader/skin-cancer-mnist-ham10000'
+  dataset = 'dinhanhx/skin-cancer-mnist-ham10000'
````
Keep indent level to 4 spaces.

Huy-Ngo commented on backend/kaggle/README.md:
```diff
@@ -91,3 +91,17 @@ Returns:
 Returns
   - `self.__img_metadata_list`
+
+`export_img_to_flask()`
+Get an image and it's info from `self.__img_metadata_list`
```
No need to duplicate doc of each methods in the class doc.

Huy-Ngo commented on backend/kaggle/datasets_manager.py:
```diff
@@ -94,11 +93,35 @@ def __init__(self, dataset_path, full_path=False):
         return self.__img_metadata_list
 
+    def export_img_to_flask(self, index=0, caption_style='HAM'):
+        """Get an image and it's info from self.__img_metadata_list
+        Parameters:
+            index : an int.
+            caption_style : a str.
+        Returns:
+            a dictionary has
+                - 'Image path'
+                - 'Host'
+                - 'Original url'
+                - 'Caption'
+        """
```
```suggestion
        """Get an image and it's info from self.__img_metadata_list

        Parameters:
            index : an int.
            caption_style : a str.

        Returns:
            a dictionary that has
                - 'path'
                - 'host'
                - 'url'
                - 'caption'
        """
```

dinhanhx replied:
Let keep it there

Huy-Ngo commented on backend/kaggle/datasets_manager.py:
```diff
@@ -94,11 +93,35 @@ def __init__(self, dataset_path, full_path=False):
     def get_img_metadata_list(self):
         return self.__img_metadata_list
 
+    def export_img_to_flask(self, index=0, caption_style='HAM'):
```
I think you can remove `get_img_metadata_list` and rename this to `get_img_metadata`

Huy-Ngo replied:
@dinhanhx should've replied to this but ok.

What I mean is delete that useless function because it's not going to be called anyway, and rename `export_img_to_flask` to `get_img_metadata`. Are you clear?

Huy-Ngo replied:
Oh just realized you were replying to my question in the big comment.

dinhanhx replied:
> @dinhanhx should've replied to this but ok.
> 
> What I mean is delete that useless function because it's not going to be called anyway, and rename `export_img_to_flask` to `get_img_metadata`. Are you clear?

Yeah I got your point.

Huy-Ngo reviewed (changes requested):
If I understand correctly, this is what I need to do, right?
```python
dataset_path = download('dinhanhx/skin-cancer-mnist-ham10000')
scm = SkinCancerMnist(dataset_path)
metadata = scm.get_img_metadata()
```

dinhanhx commented:
Yes but if you use that function, you will might have to code more to get what you need. I suggest you use new function `export_img_to_flask()`. I did put an example in the first comment of this pull request. 

For other things, I will change during tomorrow class session.

dinhanhx committed:
Class SkinCancerMnist changed
Delete func export_img_2_flaks()
Refactor func get_img_metadata()
Add func get_len_dataset()
Update corresponding part in README.md

Huy-Ngo commented:
I'll approve this if you resolve your part of #11

dinhanhx commented:
> I'll approve this if you resolve your part of #11

@Huy-Ngo please checkout this [comment](https://github.com/Huy-Ngo/skin-crawler/issues/11#issuecomment-731506307)
